lib/kube/proxy/sess.go

```diff
@@ -547,14 +552,22 @@ func (s *session) launch(isEphemeralCont bool) (returnErr error) {
 
 	eventPodMeta := request.eventPodMeta(request.context, s.sess.kubeAPICreds)
 
-	onFinished, err := s.lockedSetupLaunch(request, q, eventPodMeta)
-	if err != nil {
-		return trace.Wrap(err)
-	}
+	onFinished, err := s.lockedSetupLaunch(request, eventPodMeta)
 	defer func() {
+		onErr := func(err error) {
+			s.setTerminationErr(err)
```

Comment:
Should we check `err != nil` before running the code below, particularly the log statement?